### PR TITLE
Added userName and fullUserName to ProcessInfo

### PR DESCRIPTION
**Pull request by pvieito**
- Added ProcessInfo.userName and ProcessInfo.fullUserName as in Darwin Foundation
- Added NSFullUserName() as in Darwin Foundation
- Added underlying implementation in Core Foundation CFCopyFullUserName() for POSIX OS.

### CODE REVIEW

File: Foundation/NSPathUtilities.swift [hunk 1]
```diff
@@ -588,6 +588,10 @@ public func NSUserName() -> String {
     return userName._swiftObject
 }
 
+public func NSFullUserName() -> String {
+    NSUnimplemented()
+}
```
Comment (spevans) [inline Foundation/NSPathUtilities.swift]: Could you use something like the following to extract the gecos info?

```
public func NSFullUserName() -> String {
    let euid = _CFGetEUID()
    let uid = euid != 0 ? euid : getuid()
    var pwd = passwd()
    let maxSize = sysconf(Int32(_SC_GETPW_R_SIZE_MAX))
    let buf = UnsafeMutablePointer<Int8>.allocate(capacity: maxSize)
    defer { buf.deallocate() }
    var ptr: UnsafeMutablePointer<passwd>? = nil
    if getpwuid_r(uid, &pwd, buf, maxSize, &ptr) == 0 {
        if let gecos = pwd.pw_gecos {
            return String(cString: gecos)
        }
    }

    return ""
}
```

Comment (pvieito) [reply Foundation/NSPathUtilities.swift]: I would prefer to leave the NSFullUserName implementation for other Pull Request/Contributor.

Comment (pvieito) [reply Foundation/NSPathUtilities.swift]: Thinking about this, shouldn't this code be implemented in Core Foundation in a function similar to CFCopyUserName and made portable between multiple platforms?

Comment (spevans) [reply Foundation/NSPathUtilities.swift]: I would probably keep it in Swift and only add it into CF if the multi-platform support became too messy and a lot easier to write using the C preprocessor.

**Comment by pvieito** [comment]
Added NSFullUserName() and underlying implementation in CFCopyFullUserName() for POSIX OS.

File: CoreFoundation/Base.subproj/CFPlatform.c [hunk 2]
```diff
@@ -289,6 +289,24 @@ CF_EXPORT CFStringRef CFCopyUserName(void) {
...
+    __CFGetUGIDs(&euid, NULL);
+    struct passwd *upwd = getpwuid(euid ? euid : getuid());
+    if (upwd && upwd->pw_gecos) {
+        result = CFStringCreateWithCString(kCFAllocatorSystemDefault, upwd->pw_gecos, kCFPlatformInterfaceStringEncoding);
+    }
+#else
+#error Don't know how to compute full user name on this platform
+#endif
+    if (!result)
+        result = (CFStringRef)CFRetain(CFSTR(""));
```
Comment (parkera) [inline CoreFoundation/Base.subproj/CFPlatform.c]: Style nit here: please always use `{ }` or make one line for `if` statements.

**Comment by parkera** [comment]
Other than minor style point, looks good to me.

**Comment by pvieito** [comment]
Updated style with braces on if statements.

**Comment by parkera** [comment]
@swift-ci test and merge

**Comment by spevans** [comment]
@swift-ci test and merge

**Pull request merged**